docs/dev/oauth_for_data_integrations.md

```diff
@@ -0,0 +1,135 @@
+
+1. Navigate to the **Publishing** tab under **General**. In step 1 of the publishing flow, you receive your published client ID and secret. In step 2, you can enter additional information about your integration and see the published `app_uuid` to use below.
+
+   Save your client ID, client secret, and `app_uuid` in a secure location. 
+
+2. When opening a pull request for a **new data integration** in `integrations-extras` or `Marketplace`, use the `app_uuid` value for publishing in the `app_uuid` field of the `manifest.json` file. If the `app_uuid` values do not align, your application does not publish correctly. If you have an **existing data integration**, there is no need to update the `app_uuid`.
+
+You cannot edit a published OAuth client directly, so only go through the publishing flow when everything has been tested and is ready to go. To make updates to the OAuth client, you need to go through the publishing flow again. **The published client credentials do not appear again**.
+
+For more information about the integration publishing process, see the [Marketplace and Integrations documentation][7]. 
+
+## Add OAuth to an existing offering
+
+The process for adding an OAuth client to an existing integration is similar to what is outlined above, with some key differences.
+
+### If you have an existing data integration that’s not connected to a UI Extension
+
+Follow the [steps](#build-a-data-integration-with-oauth) above, and ensure that you open a pull request to add new uninstallation instructions to your integration tile.
+
+**Note**: There's no need to change your `app_uuid` in the manifest.json file if you have an existing data integration.
+
+### If you have an existing data integration that’s currently connected to a UI Extension (shares the same tile)
+
+Instead of creating an app, navigate to the app that includes your published UI Extension in the Developer Platform and follow the remaining [steps](#create-an-oauth-client). 
+
+Once you’ve created your data integration's OAuth client and are ready for publishing, click **Edit** on your app and navigate to the **Publishing** tab under **General**. 
+
+**Note**: There's no need to change your `app_uuid` in the manifest.json file if you have an existing data integration and/or UI Extension.
+
+### If you have a published UI Extension and want to add a data integration to the same tile
+
+Instead of creating an app, navigate to an app that includes your published UI Extension in the Developer Platform and follow the remaining [steps](#create-an-oauth-client).
+
+Open a pull request to update your existing tile with additional information about your integration—including updates to the README, image folder, and more. Additionally, update the `app_uuid` value in the manifest.json file to match the `app_uuid` value shown during the Developer Platform publishing process. Add a link to this pull request during the publishing process.
```

Comment:
This falls under "adding an integration client to an existing listing", so they would not update their `app_uuid`. The issue is that we cannot update the `uuid` of an existing app (it will create a new published app instance and all installations and other objects will not be carried over). So we will need to manually backfill these apps a well.